### ISSUE TITLE
Default Height

### ISSUE THREAD

**Issue by msalmankarim** [chrum/ngx-autosize#65]
I am trying to set default height of text-area by setting useImportant = true. The default height I want to set is 34px but it is not working. In the inspector, it is showing 47px.

![image](https://user-images.githubusercontent.com/11595290/109814380-bca1eb00-7c2e-11eb-8b16-3eb2ed2f1832.png)

![image](https://user-images.githubusercontent.com/11595290/109814759-2a4e1700-7c2f-11eb-962a-2791c7bfdbb2.png)

**Comment by chrum** [comment]
hey @msalmankarim 
Sorry for late answer :( busy busy haha

I am afraid that its not working like that. 

I think that you should use 'minRows' as the 'default height when there is no text in the text area.
But hmmmmm I am not sure how else default height should work... can you describe how you see it?
What happens when we define default height and:
- text area is empty?
- text area has some text but it takes less space than default height... should we still use default height?
- text area has lots of text and calculated size is bigger than default height

best!

**Comment by roxteddy** [comment]
Hello
It is not clear how default size works because even with minRows set to 1, the default calculated height equals to 2 lines.

**Comment by msalmankarim** [comment]
> hey @msalmankarim
> Sorry for late answer :( busy busy haha
> 
> I am afraid that its not working like that.
> 
> I think that you should use 'minRows' as the 'default height when there is no text in the text area.
> But hmmmmm I am not sure how else default height should work... can you describe how you see it?
> What happens when we define default height and:
> 
> * text area is empty?
> * text area has some text but it takes less space than default height... should we still use default height?
> * text area has lots of text and calculated size is bigger than default height
> 
> best!

Yes what I want is that to set the default height so that it can matches the other input fields in the form by default. After that if text is added and it needs more space it can resize to bigger height than the default height

**Comment by derzhimorda** [comment]
Hello everyone! 
I have the same issue, i see what that issue have not answers from 2021.. Autor, have any ideas how the fix this bug?

My case: autosize [minRows]="3", that all option what used on textarea, but if i reload page, or set for examaple tab with this textarea to not show, textarea sizing set automaticaly to lower size (42px) why?? That case with empty textarea.

**Comment by chrum** [comment]
hey,
Its a tricky one, can you create small example?

**Comment by derzhimorda** [comment]
no problem.

**Part code with autosize area**

![image](https://user-images.githubusercontent.com/18550969/215758911-645c6bbb-8df3-48dd-b3ed-fd58329bf142.png)

For example, i have tabs, one tab with textarea, if i set active tab with form, then i see textarea with style=height:63px;, after some seconds, this element set style=height:102px. This case happend if textarea form don't have content (empty). For first load (component) textarea have normal (original height (102px). This bug not critical for my project, but i am interest how to solve this..)

I am was tryied set all options for element, min-rows, max-rows, impotant height, nothing, anyway have the same problem.
Why this happend?

**Best regards!**

**Comment by GregoireKarydes** [comment]
Same issue for me

**Comment by haoqiangqiang** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
亲，邮件已收到了哦

**Comment by fernando-silva-actabl** [comment]
Use rows="1"

**Comment by haoqiangqiang** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
亲，邮件已收到了哦
